recipes/libsodium/1.0.18/conanfile.py

```diff
@@ -0,0 +1,224 @@
+from conans import ConanFile, AutoToolsBuildEnvironment, tools
+from conans.errors import ConanInvalidConfiguration
+import os
+
+
+class LibsodiumConan(ConanFile):
+    name        = "libsodium"
+    description = "A modern and easy-to-use crypto library."
+    license     = "ISC"
+    url         = "https://github.com/conan-io/conan-center-index"
+    homepage    = "https://download.libsodium.org/doc/"
+    exports_sources = ["patches/**"]
+    settings    = "os", "compiler", "arch", "build_type"
+    generators  = "cmake"
+    _source_subfolder = "source_subfolder"
+
+    options = {
+        "shared" : [True, False],
+        "fPIC": [True, False],
+        "use_soname" : [True, False],
+        "use_pie"    : [True, False],
+    }
+
+    default_options = {
+        "shared": False,
+        "fPIC": True,
+        "use_soname": True,
+        "use_pie": False,
+    }
+
+    @property
+    def android_id_str(self):
+        return "androideabi" if str(self.settings.arch) in ["armv6", "armv7"] else "android"
+
+    @property
+    def arch_id_str_compiler(self):
+        return {"x86": "i686",
+                "armv6": "arm",
+                "armv7": "arm",
+                "armv7hf": "arm",
+                "armv8": "aarch64",
+                "mips64": "mips64"}.get(str(self.settings.arch),
+                                        str(self.settings.arch))
+
+    @property
+    def is_mingw(self):
+        return self.settings.os == "Windows" and self.settings.compiler == "gcc"
+
+    @property
+    def vs_platform(self):
+        if self.settings.arch.value == "x86":
+            return "Win32"
+        return "x64"
+
+    @property
+    def vs_configuration(self):
+        configuration = ""
+        if self.options.shared:
+            configuration += "Dyn"
+        else:
+            configuration += "Static"
+        build_type = str(self.settings.build_type)
+        if build_type == "RelWithDebInfo":
+            build_type = "Release"
+        configuration += build_type
+        return configuration
+
+    @property
+    def vs_sln_folder(self):
+        if self.settings.compiler.version == "14":
+            return "vs2015"
+        elif self.settings.compiler.version == "15":
+            return "vs2017"
+        elif self.settings.compiler.version == "16":
+            return "vs2019"
+        else:
+            raise ConanInvalidConfiguration(f"Unsupported msvc version: {self.settings.compiler.version}")
+
+    @property
+    def runtime_prefix(self):
+        return str(self.settings.compiler.runtime)[:2]
+
+    def config_options(self):
+        del self.settings.compiler.libcxx
+        del self.settings.compiler.cppstd
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        extracted_dir = self.name + "-" + self.version
+        os.rename(extracted_dir, self._source_subfolder)
+
+    def build_visual(self):
```

Comment:
I believe all user-defined functions and variables should start with an underscore.